### PR TITLE
Fixed metrics path and deprecated labels for the cadvisor job

### LINKED ISSUE #3308: [Bug] Example Metric Dashboards and rules use deprecated "kubernetes_pod_name" and "container_name" labels
**Describe the bug**
The Grafana Dashboards and Prometheus AlertManager rules use the `kubernetes_pod_name` and `container_name` labels which were deprecated (see https://github.com/kubernetes/kubernetes/pull/80376).

I am currently updating the naming for the strimzi-zookeeper, strimzi-kafka, strimzi-kafka-connect Grafana Dashboards while also migrating the stats panel to Grafana v7's newest version. Let me know if there's interest in a PR.

**To Reproduce**
1. Create a kind Kubernetes v1.18.2 cluster: `kind create cluster`
2. Install the prometheus operator: `helm install prometheus stable/prometheus-operator`
3. Install Strimzi: `helm install strimzi strimzi/strimzi-kafka-operator`
4. Create a Kafka Cluster via kafka.yaml

5. Create a service monitor via kafka-sm.yaml

> Note that I have commented out the `__meta_kubernetes_endpoints_name` re-labeling as it would -for some reason- prevent metrics from being scraped at all. 

6. Open the Prometheus Operator's default Grafana Installation
7. Import the Strimzi Kafka Dashboard from <https://github.com/strimzi/strimzi-kafka-operator/blob/a040cf2f7cddcc82b0d1a731dc7bf59089537931/examples/metrics/grafana-dashboards/strimzi-kafka.json>
8. Observe that the charts display 0 as a default value or "No Data"

**Expected behavior**
The charts included in the dashboard should display the expected metrics.

**Environment (please complete the following information):**
 - Strimzi version: 0.18.0
 - Installation method: Helm
 - Kubernetes cluster: Kubernetes Kind 1.18.2
 - Prometheus Operator: prometheus-operator-8.16.1      app version: 0.38.1

**YAML files and logs**

`kafka.yaml`:

```yaml
apiVersion: kafka.strimzi.io/v1beta1
kind: Kafka
metadata:
  name: kafka-cluster
spec:
  kafkaExporter: {}
  kafka:
    version: 2.5.0
    replicas: 1
    config:
      offsets.topic.replication.factor: 1
      transaction.state.log.replication.factor: 1
      transaction.state.log.min.isr: 1
      log.message.format.version: "2.5"
    storage:
      type: ephemeral
    listeners:
      plain: {}
      tls: {}
    metrics:
      # Inspired by config from Kafka 2.0.0 example rules:
      # https://github.com/prometheus/jmx_exporter/blob/master/example_configs/kafka-2_0_0.yml
      lowercaseOutputName: true
      rules:
        # Special cases and very specific rules
        - pattern: kafka.server<type=(.+), name=(.+), clientId=(.+), topic=(.+), partition=(.*)><>Value
          name: kafka_server_$1_$2
          type: GAUGE
          labels:
            clientId: "$3"
            topic: "$4"
            partition: "$5"
        - pattern: kafka.server<type=(.+), name=(.+), clientId=(.+), brokerHost=(.+), brokerPort=(.+)><>Value
          name: kafka_server_$1_$2
          type: GAUGE
          labels:
            clientId: "$3"
            broker: "$4:$5"
        # Some percent metrics use MeanRate attribute
        # Ex) kafka.server<type=(KafkaRequestHandlerPool), name=(RequestHandlerAvgIdlePercent)><>MeanRate
        - pattern: kafka.(\w+)<type=(.+), name=(.+)Percent\w*><>MeanRate
          name: kafka_$1_$2_$3_percent
          type: GAUGE
        # Generic gauges for percents
        - pattern: kafka.(\w+)<type=(.+), name=(.+)Percent\w*><>Value
          name: kafka_$1_$2_$3_percent
          type: GAUGE
        - pattern: kafka.(\w+)<type=(.+), name=(.+)Percent\w*, (.+)=(.+)><>Value
          name: kafka_$1_$2_$3_percent
          type: GAUGE
          labels:
            "$4": "$5"
        # Generic per-second counters with 0-2 key/value pairs
        - pattern: kafka.(\w+)<type=(.+), name=(.+)PerSec\w*, (.+)=(.+), (.+)=(.+)><>Count
          name: kafka_$1_$2_$3_total
          type: COUNTER
          labels:
            "$4": "$5"
            "$6": "$7"
        - pattern: kafka.(\w+)<type=(.+), name=(.+)PerSec\w*, (.+)=(.+)><>Count
          name: kafka_$1_$2_$3_total
          type: COUNTER
          labels:
            "$4": "$5"
        - pattern: kafka.(\w+)<type=(.+), name=(.+)PerSec\w*><>Count
          name: kafka_$1_$2_$3_total
          type: COUNTER
        # Generic gauges with 0-2 key/value pairs
        - pattern: kafka.(\w+)<type=(.+), name=(.+), (.+)=(.+), (.+)=(.+)><>Value
          name: kafka_$1_$2_$3
          type: GAUGE
          labels:
            "$4": "$5"
            "$6": "$7"
        - pattern: kafka.(\w+)<type=(.+), name=(.+), (.+)=(.+)><>Value
          name: kafka_$1_$2_$3
          type: GAUGE
          labels:
            "$4": "$5"
        - pattern: kafka.(\w+)<type=(.+), name=(.+)><>Value
          name: kafka_$1_$2_$3
          type: GAUGE
        # Emulate Prometheus 'Summary' metrics for the exported 'Histogram's.
        # Note that these are missing the '_sum' metric!
        - pattern: kafka.(\w+)<type=(.+), name=(.+), (.+)=(.+), (.+)=(.+)><>Count
          name: kafka_$1_$2_$3_count
          type: COUNTER
          labels:
            "$4": "$5"
            "$6": "$7"
        - pattern: kafka.(\w+)<type=(.+), name=(.+), (.+)=(.*), (.+)=(.+)><>(\d+)thPercentile
          name: kafka_$1_$2_$3
          type: GAUGE
          labels:
            "$4": "$5"
            "$6": "$7"
            quantile: "0.$8"
        - pattern: kafka.(\w+)<type=(.+), name=(.+), (.+)=(.+)><>Count
          name: kafka_$1_$2_$3_count
          type: COUNTER
          labels:
            "$4": "$5"
        - pattern: kafka.(\w+)<type=(.+), name=(.+), (.+)=(.*)><>(\d+)thPercentile
          name: kafka_$1_$2_$3
          type: GAUGE
          labels:
            "$4": "$5"
            quantile: "0.$6"
        - pattern: kafka.(\w+)<type=(.+), name=(.+)><>Count
          name: kafka_$1_$2_$3_count
          type: COUNTER
        - pattern: kafka.(\w+)<type=(.+), name=(.+)><>(\d+)thPercentile
          name: kafka_$1_$2_$3
          type: GAUGE
          labels:
            quantile: "0.$4"
  zookeeper:
    replicas: 3
    storage:
      type: ephemeral
    metrics:
      # Inspired by Zookeeper rules
      # https://github.com/prometheus/jmx_exporter/blob/master/example_configs/zookeeper.yaml
      lowercaseOutputName: true
      rules:
        # replicated Zookeeper
        - pattern: "org.apache.ZooKeeperService<name0=ReplicatedServer_id(\\d+)><>(\\w+)"
          name: "zookeeper_$2"
          type: GAUGE
        - pattern: "org.apache.ZooKeeperService<name0=ReplicatedServer_id(\\d+), name1=replica.(\\d+)><>(\\w+)"
          name: "zookeeper_$3"
          type: GAUGE
          labels:
            replicaId: "$2"
        - pattern: "org.apache.ZooKeeperService<name0=ReplicatedServer_id(\\d+), name1=replica.(\\d+), name2=(\\w+)><>(Packets\\w+)"
          name: "zookeeper_$4"
          type: COUNTER
          labels:
            replicaId: "$2"
            memberType: "$3"
        - pattern: "org.apache.ZooKeeperService<name0=ReplicatedServer_id(\\d+), name1=replica.(\\d+), name2=(\\w+)><>(\\w+)"
          name: "zookeeper_$4"
          type: GAUGE
          labels:
            replicaId: "$2"
            memberType: "$3"
        - pattern: "org.apache.ZooKeeperService<name0=ReplicatedServer_id(\\d+), name1=replica.(\\d+), name2=(\\w+), name3=(\\w+)><>(\\w+)"
          name: "zookeeper_$4_$5"
          type: GAUGE
          labels:
            replicaId: "$2"
            memberType: "$3"
        # standalone Zookeeper
        - pattern: "org.apache.ZooKeeperService<name0=StandaloneServer_port(\\d+)><>(\\w+)"
          type: GAUGE
          name: "zookeeper_$2"
        - pattern: "org.apache.ZooKeeperService<name0=StandaloneServer_port(\\d+), name1=InMemoryDataTree><>(\\w+)"
          type: GAUGE
          name: "zookeeper_$2"
  entityOperator:
    topicOperator: {}
```

`kafka-sm.yaml`:

```yaml
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: kafka-cluster-service-monitor
  labels:
    release: prometheus
spec:
  selector:
    matchExpressions:
      - { key: strimzi.io/kind, operator: In, values: [Kafka, KafkaConnect] }
  namespaceSelector:
    matchNames:
      - default
  endpoints:
    - port: tcp-prometheus
      honorLabels: true
      interval: 10s
      scrapeTimeout: 10s
      path: /metrics
      scheme: http
      relabelings:
        # - sourceLabels: [__meta_kubernetes_endpoints_name]
        #   separator: ;
        #   regex: prometheus-kube-state-metrics
        #   replacement: $1
        #   action: keep
        - separator: ;
          regex: __meta_kubernetes_service_label_(.+)
          replacement: $1
          action: labelmap
        - sourceLabels: [__meta_kubernetes_namespace]
          separator: ;
          regex: (.*)
          targetLabel: namespace
          replacement: $1
          action: replace
        - sourceLabels: [__meta_kubernetes_namespace]
          separator: ;
          regex: (.*)
          targetLabel: kubernetes_namespace
          replacement: $1
          action: replace
        - sourceLabels: [__meta_kubernetes_service_name]
          separator: ;
          regex: (.*)
          targetLabel: kubernetes_name
          replacement: $1
          action: replace
        - sourceLabels: [__meta_kubernetes_pod_node_name]
          separator: ;
          regex: (.*)
          targetLabel: node_name
          replacement: $1
          action: replace
        - sourceLabels: [__meta_kubernetes_pod_host_ip]
          separator: ;
          regex: (.*)
          targetLabel: node_ip
          replacement: $1
          action: replace

    #### job_name: node-exporter
    - port: tcp-prometheus
      honorLabels: true
      interval: 10s
      scrapeTimeout: 10s
      path: /metrics
      scheme: http
      relabelings:
        - sourceLabels: [__meta_kubernetes_endpoints_name]
          separator: ;
          regex: prometheus-node-exporter
          replacement: $1
          action: keep
        - separator: ;
          regex: __meta_kubernetes_service_label_(.+)
          replacement: $1
          action: labelmap
        - sourceLabels: [__meta_kubernetes_namespace]
          separator: ;
          regex: (.*)
          targetLabel: namespace
          replacement: $1
          action: replace
        - sourceLabels: [__meta_kubernetes_namespace]
          separator: ;
          regex: (.*)
          targetLabel: kubernetes_namespace
          replacement: $1
          action: replace
        - sourceLabels: [__meta_kubernetes_service_name]
          separator: ;
          regex: (.*)
          targetLabel: kubernetes_name
          replacement: $1
          action: replace
        - sourceLabels: [__meta_kubernetes_pod_node_name]
          separator: ;
          regex: (.*)
          targetLabel: node_name
          replacement: $1
          action: replace
        - sourceLabels: [__meta_kubernetes_pod_host_ip]
          separator: ;
          regex: (.*)
          targetLabel: node_ip
          replacement: $1
          action: replace
```

### PR DESCRIPTION
**Pull request by ppatierno**
Fixed dashboards using the right label

Signed-off-by: Paolo Patierno <ppatierno@live.com>

### Type of change

_Select the type of your PR_

- Bugfix

### Description

This PR fixes #3308.
Memory and CPU usage metrics are not working anymore on Kafka and ZooKeeper dashboards due to change on the path where cadvisor exposes the metrics and some deprecated and dropped labels on the related metrics.
Following the current not working metrics for Kafka and ZooKeeper dashboards.

without_metrics

and here the working ones using this PR.

with_metrics

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging

### CODE REVIEW

**Comment by tombentley** [comment]
The referenced [Kubernetes PR](https://github.com/kubernetes/kubernetes/pull/80376) says 

> Duplicate labels were included in the 1.14 and 1.15 releases

Merging this PR would break support for Kubernetes 1.11–1.13. I know we're on the verge of dropping support for those versions ourselves, but we've not yet done so.

**Comment by ppatierno** [comment]
We have always said that they are examples right? We don't provide support for them so why are not breaking compatibility with something we don't actually support. What are you proposing? Having the metrics working only on older Kubernetes version and having them broken on new ones?

**Comment by tombentley** [comment]
I'd forgotten that we've always said these were examples, rather than anything supported. However, it's probably worth mentioning in the changelog that we're dropping support for metrics in those older versions, and linking to this PR, so that it's more clear that we've done this on purpose, and users on older versions at least know how to fix things.

File: CHANGELOG.md [hunk 1]
```diff
@@ -37,6 +37,13 @@ In Strimzi 0.12.0, the `v1alpha1` versions of the following resources have been
 In the next release, the `v1alpha1` versions of these resources will be removed. 
 Please follow the guide for upgrading the resources: https://strimzi.io/docs/operators/latest/full/deploying.html#assembly-upgrade-resources-str.
 
+#### Removal deprecated cadvisor metric labels
+
+The `pod_name` and `container_name` labels provided on the cadvisor metrics are now just `pod` and `container` starting from Kubernetes 1.16.
+We removed the old ones from the Prometheus scraping configuration/alerts and on the Kafka and ZooKeeper dashboard as well.
+It means that the charts related to memory and CPU usage are not going to work on Kuvbernetes version previous 1.14.
+For more information see this Kubernetes PR https://github.com/kubernetes/kubernetes/pull/80376
```
Comment (tombentley) [inline CHANGELOG.md:45]: It would be better to link directly to _this_ PR, that way they can immediately see what got changed.

Comment (ppatierno) [reply CHANGELOG.md:45]: makes more sense ;-)

**Pull request merged**